### PR TITLE
Domain tests

### PR DESCRIPTION
**Pull request by dstenroejl**
## Description

Implement first domain test,

Is related to PR in dh3-environments. Current PR must be closed first, then the one in dh3-environments.
https://github.com/Energinet-DataHub/dh3-environments/pull/327

## References

- #556 
- https://app.zenhub.com/workspaces/mighty-ducks---the-outlaws-6193fe815d79fc0011e741b1/issues/energinet-datahub/the-outlaws/694

### CODE REVIEW

File: source/dotnet/Services/DomainTests/DomainTests.csproj [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  <PropertyGroup>
+    <IsPackable>false</IsPackable>
+    <AssemblyName>Energinet.DataHub.Wholesale.DomainTests</AssemblyName>
+    <RootNamespace>Energinet.DataHub.Wholesale.DomainTests</RootNamespace>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Energinet.DataHub.Core.FunctionApp.TestCommon" Version="3.5.0" />
+    <PackageReference Include="Energinet.DataHub.Wholesale.Client" Version="3.4.0-alpha-191" />
```
Comment (dstenroejl) [inline source/dotnet/Services/DomainTests/DomainTests.csproj:11]: Must use release version

File: source/dotnet/Services/DomainTests/WebApiTests.cs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+                // Act
+                var batchResult = await Fixture.WholesaleClient.GetBatchAsync(Fixture.Configuration.ExistingBatchId);
+
+                // Assert
+                batchResult.Should().NotBeNull();
+                batchResult!.BatchNumber.Should().Be(Fixture.Configuration.ExistingBatchId);
+            }
+
+            [DomainFact(Skip = "Enable when When_UsingRawHttpClientCreatingBatch_Then_BatchIsEventuallyExecuting is removed")]
```
Comment (dstenroejl) [inline source/dotnet/Services/DomainTests/WebApiTests.cs:114]: Enable this test when Wholesale client package and new batch controller has been released

File: source/dotnet/Services/DomainTests/WebApiTests.cs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                    },
+                    DefaultTimeout,
+                    DefaultDelay);
+
+                isExecuting.Should().BeTrue();
+            }
+
+            // TODO: Remove when new WholesaleClient.CreateBatchAsync is released.
+            [DomainFact]
+            public async Task When_UsingRawHttpClientCreatingBatch_Then_BatchIsEventuallyExecuting()
```
Comment (dstenroejl) [inline source/dotnet/Services/DomainTests/WebApiTests.cs:140]: Remove this test when Wholesale client package and new batch controller has been released

**Review by lasrinnil — DISMISSED** [review]
Everything looks good! 

Maybe a lot of the stuff in the `Fixtures` folder could be a contender to be part of the cool test.common NuGet package :)

**Review by johevemi — APPROVED** [review]

**Review by johevemi — APPROVED** [review]

**Review by Renetnielsen — APPROVED** [review]

**Pull request merged**